models/reviews.model.js

```diff
@@ -45,29 +44,73 @@ exports.updateReviewById = (reviewId, updateVotes) => {
     });
 };
 
-exports.fetchAllReviews = () => {
-  return db
-    .query(
-      `SELECT reviews.owner,
-      reviews.title,
-      reviews.review_id,
-      reviews.category,
-      reviews.review_img_url,
-      reviews.created_at,
-      reviews.votes,
-      COUNT (comments.body) AS comment_count
+exports.fetchAllReviews = (
+  category,
+  sort_by = 'created_at',
+  order_By = 'DESC'
+) => {
+  const order_by = order_By.toUpperCase();
+  const availableSortColumns = [
+    'owner',
+    'title',
+    'review_id',
+    'category',
+    'created_at',
+    'votes',
+    'comment_count',
+  ];
+  const availableOrderBy = ['ASC', 'DESC'];
+
+  if (
+    !availableOrderBy.includes(order_by) ||
+    !availableSortColumns.includes(sort_by)
+  ) {
+    return Promise.reject({ status: 400, msg: 'Wrong input' });
+  }
+
+  let queryStr = `SELECT 
+  reviews.owner,
+  reviews.title,
+  reviews.review_id,
+  reviews.category,
+  reviews.review_img_url,
+  reviews.created_at,
+  reviews.votes,
+  COUNT (comments.body) AS comment_count 
   FROM reviews
-  LEFT JOIN comments
-  ON reviews.review_id=comments.review_id
-  GROUP BY reviews.review_id
-  ORDER BY created_at DESC`
-    )
-    .then(({ rows }) => {
-      return rows.map((review) => {
-        review.comment_count = Number(review.comment_count);
-        return review;
-      });
-    });
+  LEFT JOIN comments 
+    ON reviews.review_id=comments.review_id `;
+
+  const queryVals = [];
+  if (!category) {
+    queryStr += ` 
+    GROUP BY reviews.review_id
+    ORDER BY ${sort_by} ${order_by}`;
+  } else {
+    queryStr += ` 
+    WHERE reviews.category=$1 
+    GROUP BY reviews.review_id
+    ORDER BY ${sort_by} ${order_by}`;
+    queryVals.push(category);
+  }
+  const tableQuery = db.query(queryStr, queryVals);
+  const allCategories = db.query(`SELECT * FROM categories`);
+  return Promise.all([tableQuery, allCategories]).then(
+    ([resultTable, resultCategories]) => {
+      const endpointQuery = resultTable.rows;
+      const categoryCheck = resultCategories.rows.map((obj) => obj.slug);
```

Comment:
I would prefer you did this check earlier, as by this point, you could have already been subject to SQL injection.

Reply:
I don't quite get which check should I refactor. Could you please specify?